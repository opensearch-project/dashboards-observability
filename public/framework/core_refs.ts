/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import DSLService from 'public/services/requests/dsl';
import {
  ApplicationStart,
  ChromeStart,
  CoreStart,
  HttpStart,
  IToasts,
  OverlayStart,
  SavedObjectsClientContract,
} from '../../../../src/core/public';
import { DashboardStart } from '../../../../src/plugins/dashboard/public';
import { DataSourcePluginStart } from '../../../../src/plugins/data_source/public';
import PPLService from '../services/requests/ppl';

class CoreRefs {
  private static _instance: CoreRefs;

  public core?: CoreStart;
  public http?: HttpStart;
  public savedObjectsClient?: SavedObjectsClientContract;
  public pplService?: PPLService;
  public dslService?: DSLService;
  public toasts?: IToasts;
  public chrome?: ChromeStart;
  public application?: ApplicationStart;
  public queryAssistEnabled?: boolean;
  public summarizeEnabled?: boolean;
  public dashboard?: DashboardStart;
  public dashboardProviders?: unknown;
  public overlays?: OverlayStart;
  public dataSource?: DataSourcePluginStart;
  private constructor() {
    // ...
  }

  public static get Instance() {
    // Do you need arguments? Make it a regular static method instead.
    return this._instance || (this._instance = new this());
  }
}

export const coreRefs = CoreRefs.Instance;
