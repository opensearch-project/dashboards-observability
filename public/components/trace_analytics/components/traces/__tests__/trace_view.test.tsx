/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { TraceView } from '..';
import { coreRefs } from '../../../../../framework/core_refs';

describe('Trace view component', () => {
  configure({ adapter: new Adapter() });

  it('renders trace view', () => {
    const { http, chrome } = coreRefs;
    const wrapper = shallow(
      <TraceView
        http={http!}
        chrome={chrome!}
        parentBreadcrumbs={[{ text: 'test', href: 'test#/' }]}
        traceId="test"
        mode="data_prepper"
        dataSourceMDSId={[{ id: '', label: '' }]}
        attributesFilterFields={[]}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
