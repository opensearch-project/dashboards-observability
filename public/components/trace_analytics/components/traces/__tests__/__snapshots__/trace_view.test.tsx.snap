// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trace view component renders trace view 1`] = `
<Fragment>
  <EuiPage>
    <EuiPageBody>
      <EuiFlexItem>
        <EuiText
          size="s"
        >
          <h1
            className="overview-content"
          >
            test
          </h1>
        </EuiText>
      </EuiFlexItem>
      <EuiFlexGroup
        alignItems="stretch"
        gutterSize="s"
      >
        <EuiFlexItem
          grow={5}
        >
          <EuiPanel>
            <PanelTitle
              title="Overview"
            />
            <EuiHorizontalRule
              margin="m"
            />
            <EuiFlexGroup>
              <EuiFlexItem>
                <EuiFlexGroup
                  direction="column"
                >
                  <EuiFlexItem
                    grow={false}
                  >
                    <EuiText
                      className="overview-title"
                    >
                      Trace ID
                    </EuiText>
                  </EuiFlexItem>
                  <EuiFlexItem
                    grow={false}
                  >
                    <EuiText
                      className="overview-title"
                    >
                      Trace group name
                    </EuiText>
                    <EuiText
                      className="overview-content"
                      size="s"
                    >
                      -
                    </EuiText>
                  </EuiFlexItem>
                </EuiFlexGroup>
              </EuiFlexItem>
              <EuiFlexItem>
                <EuiFlexGroup
                  direction="column"
                >
                  <EuiFlexItem
                    grow={false}
                  >
                    <EuiText
                      className="overview-title"
                    >
                      Latency
                    </EuiText>
                    <EuiText
                      className="overview-content"
                      size="s"
                    />
                  </EuiFlexItem>
                  <EuiFlexItem
                    grow={false}
                  >
                    <EuiText
                      className="overview-title"
                    >
                      Last updated
                    </EuiText>
                    <EuiText
                      className="overview-content"
                      size="s"
                    />
                  </EuiFlexItem>
                </EuiFlexGroup>
              </EuiFlexItem>
              <EuiFlexItem>
                <EuiFlexGroup
                  direction="column"
                >
                  <EuiFlexItem
                    grow={false}
                  >
                    <EuiText
                      className="overview-title"
                    >
                      Errors
                    </EuiText>
                    <EuiText
                      className="overview-content"
                      size="s"
                    >
                      -
                    </EuiText>
                  </EuiFlexItem>
                </EuiFlexGroup>
              </EuiFlexItem>
            </EuiFlexGroup>
          </EuiPanel>
        </EuiFlexItem>
        <EuiFlexItem
          grow={3}
        >
          <ServiceBreakdownPanel
            data={Array []}
            isLoading={false}
          />
        </EuiFlexItem>
      </EuiFlexGroup>
      <EuiSpacer
        size="m"
      />
      <EuiFlexGroup>
        <EuiFlexItem>
          <SpanDetailPanel
            colorMap={Object {}}
            data={
              Object {
                "gantt": Array [],
                "ganttMaxX": 0,
                "table": Array [],
              }
            }
            dataSourceMDSId=""
            dataSourceMDSLabel=""
            http={[MockFunction]}
            mode="data_prepper"
            setData={[Function]}
            traceId="test"
          />
        </EuiFlexItem>
      </EuiFlexGroup>
      <EuiSpacer />
      <EuiPanel>
        <EuiFlexGroup>
          <EuiFlexItem>
            <PanelTitle
              title="Payload"
            />
          </EuiFlexItem>
        </EuiFlexGroup>
        <EuiHorizontalRule
          margin="m"
        />
      </EuiPanel>
      <EuiSpacer />
      <ServiceMap
        idSelected="latency"
        isServicesDataLoading={false}
        page="traceView"
        serviceMap={Object {}}
        setIdSelected={[Function]}
      />
    </EuiPageBody>
  </EuiPage>
</Fragment>
`;
